src/defu.ts

```diff
@@ -18,17 +23,15 @@ function _defu<T>(
   const object = Object.assign({}, defaults);
 
   for (const key in baseObject) {
-    if (key === "__proto__" || key === "constructor") {
-      continue;
-    }
-
     const value = baseObject[key];
 
-    if (value === null || value === undefined) {
-      continue;
-    }
-
-    if (merger && merger(object, key, value, namespace)) {
```

Comment:
The current syntax is actually intentional in order to avoid accessing value before key check and also keep separate sections more readable. In general when contributing to an open source project, please consider avoiding style changes like this unless necessary or separate it to another PR 🙏🏼